lib/document.js

```diff
@@ -2217,19 +2217,26 @@ Document.prototype[documentModifiedPaths] = Document.prototype.modifiedPaths;
 
 Document.prototype.isModified = function(paths, modifiedPaths) {
   if (paths) {
-    const directModifiedPaths = Object.keys(this.$__.activePaths.getStatePaths('modify'));
-    if (directModifiedPaths.length === 0) {
-      return false;
-    }
-
     if (!Array.isArray(paths)) {
       paths = paths.indexOf(' ') === -1 ? [paths] : paths.split(' ');
     }
+
+    const directModifiedPathsObj = this.$__.activePaths.states.modify;
+    if (directModifiedPathsObj == null) {
+      return false;
+    }
+    for (const path of paths) {
+      if (directModifiedPathsObj.hasOwnProperty(path)) {
```

Comment:
Should use Object.prototype.hasOwnProperty. but still dangerous if the object is not using NullObject.